touchforms/backend/xformserver.py

```diff
@@ -259,10 +259,9 @@ def handle_request(content, server):
             result = xfsess.evaluate_xpath(content['xpath'])
             return {"output": result['output'], "status": result['status']}
         elif action == xformplayer.Actions.SYNC_USER_DB:
-            ensure_required_params(['username', 'hq_auth'], action, content)
-            trimmed_username = content['username'][:content['username'].index('.')]
-            result = touchcare.force_ota_restore(trimmed_username, auth=content['hq_auth'])
-            return result
+            ensure_required_params(['username', 'hq_auth', 'domain'], action, content)
+            trimmed_username = content['username'].split('@')[0]
```

Comment:
au contraire mr pride, this does in fact work.  i didn't know this until you mentioned it and i looked into why it was succeeding for me. i think when we supply a user with `brudolph@wrong-domain` it uses that wrong domain and then verifies incorrectly. the view code gets the user from the hq_auth so the user isn't incorrect. so that's why it seems that `brudolph@correct-domain` works